base_layer/wallet/src/base_node_service/monitor.rs

```diff
@@ -198,24 +216,27 @@ where
         Ok(())
     }
 
-    async fn update_state(&self, new_state: BaseNodeState) {
+    // returns true if a new block, otherwise false
+    async fn update_state(&self, new_state: BaseNodeState) -> bool {
         let mut lock = self.state.write().await;
-        let (new_block_detected, height) = match (new_state.chain_metadata.clone(), lock.chain_metadata.clone()) {
+        let (new_block_detected, height, hash) = match (new_state.chain_metadata.clone(), lock.chain_metadata.clone()) {
```

Comment:
```suggestion
        let (new_block_detected, connected_base_node_change_detected , height, hash) = match (new_state.chain_metadata.clone(), lock.chain_metadata.clone()) {
```

Reply:
It would be better to open a ticket for this, so that we add the handling of this event in the correct places.

I don't think it's necessary, because I believe the logic of the code is to send a `Default` state when the base node changes. This will not publish the NewBlockEvent, but when we do get a chain metadata from the new base node, if it has the same block hash, there is no need to update, but if it changes, it will trigger the appropriate jobs

Reply:
Yeah, evaluating the block hash will sort this out